jsettlers.tools/src/main/java/jsettlers/buildingcreator/editor/BuildingCreatorApp.java

```diff
@@ -68,81 +99,195 @@ public class BuildingCreatorApp implements IMapInterfaceListener, Runnable {
 	private ToolType tool = ToolType.SET_BLOCKED;
 	private JLabel positionDisplayer;
 	private JFrame window;
+        private IMapInterfaceConnector mapInterfaceConnector;
+
+        private Timer timer;
 
 	@Override
 	public void run() {
 		try {
 			EBuildingType type = askType();
+                        if (type == null) {
+                            // user pressed Cancel
+                            return;
+                        }
 			BuildingVariant variant = askVariant(type);
+                        if (variant == null) {
+                            // user pressed Cancel
+                            return;
+                        }
 
 			definition = new BuildingDefinition(variant);
 			map = new BuildingtestMap(definition);
-			for (int x = 0; x < map.getWidth(); x++) {
-				for (int y = 0; y < map.getHeight(); y++) {
-					reloadColor(new ShortPoint2D(x, y));
-				}
-			}
 
-			IMapInterfaceConnector connector = startMapWindow();
-			connector.addListener(this);
+                        reloadMapColor();
+
+			mapInterfaceConnector = startMapWindow();
+			mapInterfaceConnector.addListener(this);
 
-			JPanel menu = generateMenu();
+			JComponent menu = generateMenu();
 
-			window = new JFrame("Edit " + variant.toString());
+			window = new JFrame("Edit " + variant);
 			window.setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
-			window.add(menu);
+                        window.setLayout(new BorderLayout());
+			window.add(new JScrollPane(menu), BorderLayout.CENTER);
 			window.pack();
 			window.setVisible(true);
 
-			connector.fireAction(new Action(EActionType.TOGGLE_DEBUG));
+			mapInterfaceConnector.fireAction(new Action(EActionType.TOGGLE_DEBUG));
+                        
 		} catch (JSettlersLookAndFeelExecption e) {
 			throw new RuntimeException(e);
 		}
-	}
 
-	private JPanel generateMenu() {
-		JPanel menu = new JPanel();
-		menu.setLayout(new BoxLayout(menu, BoxLayout.Y_AXIS));
-		menu.add(createToolChangeBar());
-
-		JButton xmlButton = new JButton("show xml data");
-		xmlButton.addActionListener(e -> showXML());
-		menu.add(xmlButton);
-		positionDisplayer = new JLabel();
-		menu.add(positionDisplayer);
-		return menu;
+	}
+        
+	private JComponent generateMenu() {
+            JTabbedPane menu = new JTabbedPane();
+                
+            menu.add("Area", createToolChangeBar());
+
+            // do we have a military building?
+            switch (definition.getBuilding().getType()) {
+                case BIG_TOWER:
+                case CASTLE:
+                case LOOKOUT_TOWER:
```

Comment:
lookout towers can't have soldiers inside. Generally speaking every building can't have occupierplaces so I don't see a strong reason to limit it to military buildings

Reply:
"every building can't have occupierplaces" probably means that potentially every building can have OccupierPlaces? If that is the case, I will just drop that limitation.